salt/master.py

```diff
@@ -1269,9 +1269,12 @@ def _file_recv(self, load):
             return {}
         load.pop('tok')
 
+        # Join path
+        sep_path = os.sep.join(load['path'])
```

Comment:
you lost me. 
If the tuple is an issue I can try to pass it from the minion in that way but I don't really see the point.
By any means, I'm not trying to disagree just to understand. I'm sure I'm missing some info on how this works.